include/podio/SchemaEvolution.h

```diff
@@ -2,13 +2,147 @@
 #define PODIO_SCHEMAEVOLUTION_H
 
 #include <cstdint>
+#include <functional>
+#include <string>
+#include <unordered_map>
+#include <vector>
 
 namespace podio {
 
 enum class Backend { ROOT, SIO };
 
+/// The type used for schema versions throughout
 using SchemaVersionT = uint32_t;
 
+struct CollectionReadBuffers;
+
+/**
+ * The SchemaEvolution registry holds evolution functions that allow to
+ * transform CollectionReadBuffers of known datatypes from a previous schema
+ * version to the current schema version. From the evolved buffers it is then
+ * possible to create collections.
+ *
+ * It is implemented as a singleton that is populated at the time shared
+ * datamodel libraries (or their schema evolution libraries) are loaded. It is
+ * assumed that this happens early on in the startup of any application, such
+ * that the registration still happens on a single thread. After this
+ * initialization evolutions can be done from multiple threads.
+ */
+class SchemaEvolution {
```

Comment:
not sure why you talk about SchemaEvolution and SchemaEvolutionRegistry interchangeably. Maybe one should just make a global (and stupid) rename?

Reply:
Removed the registry bits again. They were leftovers from when it was not yet entirely clear to me what the name should be.